packages/safe-core-sdk/src/utils/signatures/index.ts

```diff
@@ -13,32 +13,17 @@ export function generatePreValidatedSignature(ownerAddress: string): SafeSignatu
   return new EthSignSignature(ownerAddress, signature)
 }
 
-export async function generateSignature(
-  ethers: any,
-  signer: Signer,
-  hash: string
-): Promise<EthSignSignature> {
-  const signerAddress = await signer.getAddress()
-  const messageArray = ethers.utils.arrayify(hash)
-  let signature = await signer.signMessage(messageArray)
-  const hasPrefix = isTxHashSignedWithPrefix(hash, signature, signerAddress)
-  let signatureV = parseInt(signature.slice(-2), 16)
-  switch (signatureV) {
-    case 0:
-    case 1:
-      signatureV += 31
-      break
-    case 27:
-    case 28:
-      if (hasPrefix) {
-        signatureV += 4
-      }
-      break
-    default:
-      throw new Error('Invalid signature')
-  }
-  signature = signature.slice(0, -2) + signatureV.toString(16)
-  return new EthSignSignature(signerAddress, signature)
+function isWalletConnectProvider(signer: Signer) {
+  return !!((signer?.provider as any)?.provider?.wc)
+}
+
+async function generateWalletConnectSignature(ethers: any, hash: any, signer: Signer): Promise<string> {
+  const prefix = ethers.utils.toUtf8Bytes(`\x19Ethereum Signed Message:\n${hash.length}`)
```

Comment:
Doesn't this depend on the connected wallet? With which wallets did you test it?

Reply:
I tested with MetaMask directly and with MetaMask mobile via WalletConnect

Reply:
I would recommend to test also with other wallets, as MetaMask doesn't follow the standard when it comes to `eth_sign`. But this should not be an issues since the signatures is also valid without the prefix, but in this case you don't need to add `+4` to `v` ... So I would prefer if we don't add the prefix manually and rather check if it was used and adjust `v` according to that